src/coord/src/coord.rs

```diff
@@ -867,6 +903,93 @@ impl<S: Append + 'static> Coordinator<S> {
             .unwrap();
     }
 
+    /// Attempts to commit all pending write transactions in a group commit. If the timestamp
+    /// chosen for the writes is not ahead of `now`, then we can execute and commit the writes
+    /// immediately. Otherwise we must wait for `now` to advance past the timestamp chosen for the
+    /// writes.
+    async fn try_group_commit(&mut self, timestamp: Option<Timestamp>) {
+        if self.pending_writes.is_empty() {
+            return;
+        }
+
+        let timestamp = if let Some(timestamp) = timestamp {
+            timestamp
+        } else {
+            self.peek_local_ts()
```

Comment:
I think we should call peek_local_ts each time, and not pass a timestamp to the next invocation. If the timestamp oracle has advanced during the sleep then it's ok for us to need to sleep again.

Reply:
Yeah good point, I've made this change.